tutorials/themes/tutorials-theme/static/searchtools.js

```diff
@@ -773,86 +820,135 @@ var Search = {
+
+    // Find the first image in the tutorial and use that as a thumbnail for
+    // the search results.
+    // TODO: in the future, we may want to make the thumbnail image a
+    // customizable setting for each tutorial
+    var imageLineIndex = textLower.indexOf('.. image::');
+    if (imageLineIndex != -1) {
+      var imageLineEndIndex = textLower.indexOf("\n", imageLineIndex);
+      var imageLine = text.substr(imageLineIndex,
+        imageLineEndIndex-imageLineIndex);
+      var imageName = imageLine.split('nboutput/')[1];
+      var imageURL = "_images/" + imageName;
+    } else {
+      var imageURL = "_static/default_thumbnail.png";
     }
 
-    //content to displayed on the placards on search queries
-    else{
-      var startPoint = textLower.indexOf('====',0);
-      var start = startPoint;
-      var sentenceStart = 0;  
-      $.each(keywords, function() {
-        var finderStart = startPoint;
-        var flag = 0;
-        var firstFound =-1;
-      
-        while(flag==0 && sentenceStart!=80){
-          var i = textLower.indexOf(this.toLowerCase(),finderStart);
-          if (i > -1)
-            start = i;
-          else if (i==-1 && firstFound!=-1)
-            {start = firstFound;
-             sentenceStart=0;
-             flag =1;
-             break;}
-            
-          for( sentenceStart = 0; sentenceStart<80; sentenceStart++){ 
-            if((textLower.charAt(start - sentenceStart) == '>') || (textLower.charAt(start - sentenceStart) == '<')) {
-              finderStart = start+1;
-              if(finderStart == startPoint)
-              {firstFound = start;}
-              break;
-            }
-            else if(textLower.charAt(start - sentenceStart) == '.') {
-              flag = 1;
-              break;}
-          }
-          }
-        });
-      start = Math.max(start - sentenceStart, 0);
-      var trimmedText = ((start > 0) ? '...' : '') + $.trim(textLower.substr(start, 350)) + ((start + 350 - textLower.length) ? '...' : '');
+    function cleanupRSTString(rstString) {
+      var newStr;
+
+      // Replace continuous whitespace with single whitespace
+      newStr = rstString.replace(/(\s{2,})/gm, " ");
+      newStr = newStr.replace(/(\r\n|\n|\r)/gm, " ");
+
+      // Replace heading underlines and newlines:
+      newStr = newStr.replace(/(\s[=]{3,}\s)/gm, ": ");
+      newStr = newStr.replace(/(\s[\-]{3,}\s)/gm, ": ");
+
+      // Remove Sphinx directives:
+      newStr = newStr.replace(/(.. [a-zA-Z0-9\-]+::)/gm, "");
+      newStr = newStr.replace(/(:[a-zA-Z0-9\-]+:)/gm, "");
+
+      // Remove URLs:
+      newStr = newStr.replace(/(?:www|https?)[^\s]+/gm, "");
+
+      // Remove RST formatting:
+      newStr = newStr.replace(/(`)/gm, "");
+
+      return newStr;
+    }
+
+    // Start looking at search text below the main header
+    var startIdx = textLower.indexOf('====\n', 0) + 5;
```

Comment:
Also, I think we should update startIdx so that we retrieve more informatic text about the tutorial. 

<img width="1507" alt="Screenshot 2019-09-16 at 2 44 22 PM" src="https://user-images.githubusercontent.com/24625562/64946956-9d27ce80-d891-11e9-9cee-1fa7f36e0a18.png">